tests/basic/pure_test.go

```diff
@@ -4238,6 +4238,248 @@ var _ = Describe("{CreateNewPoolsWhenFadaFbdaVolumeCreationInProgress}", func()
+			faParams["fs"] = "ext4"
+
+			var allowVolExpansionFA bool = true
+			// create storage class for FADA volumes
+			err = CreateFlashStorageClass(fadaScName,
+				"pure_block",
+				v1.PersistentVolumeReclaimDelete,
+				faParams, nil, &allowVolExpansionFA,
+				storageApi.VolumeBindingImmediate,
+				nil)
+			log.FailOnError(err, fmt.Sprintf("Failed to create storage class [%v] ", fadaScName))
+			log.InfoD("Storage class [%s] for FADA is created", fadaScName)
+
+			fbParams := make(map[string]string)
+			fbParams["pure_export_rules"] = "*(rw)"
+			// create storage class for FADA volumes
+			err = CreateFlashStorageClass(fbdaScName,
+				"pure_file",
+				v1.PersistentVolumeReclaimDelete,
+				fbParams, nil, &allowVolExpansionFA,
+				storageApi.VolumeBindingImmediate,
+				nil)
+			log.FailOnError(err, fmt.Sprintf("Failed to create storage class [%v] ", fbdaScName))
+			log.InfoD("Storage class [%s] for FBDA is created", fbdaScName)
+		})
+
+		//CreatePVC will create a pvc with given name, storage class, size and namespace
+		createPVC := func(pvcName string, scName string, pvcSize string, ns string) (*v1.PersistentVolumeClaim, error) {
+			size, err := resource.ParseQuantity(pvcSize)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse pvc size : %s", pvcSize)
+			}
+			pvcClaimSpec := k8s.MakePVC(size, ns, pvcName, scName)
+			pvc, err := k8sCore.CreatePersistentVolumeClaim(pvcClaimSpec)
+			return pvc, err
+		}
+		//createAndAppendPVC will call createPVC with given name, storage class, size and namespace
+		createAndAppendPVC := func(appName string, scName string, namespace string, x int) {
+			pvcName := fmt.Sprintf("%s-%d", appName, x)
+			pvc, err := createPVC(pvcName, scName, "10", namespace)
+			log.FailOnError(err, "Failed to create PVC [%s] in namespace [%s]", pvcName, namespace)
+			log.InfoD("PVC [%s] created in namespace [%s]", pvc.Name, namespace)
+
+		}
+		//createNameSpace will create a namespace with given name and label
+		createNameSpace := func(namespace string) error {
+			nsSpec := &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
+			_, err := k8sCore.CreateNamespace(nsSpec)
+			return err
+		}
+
+		stepLog = "Create Namespace and then create 10 PVC each with respective storage class parallely"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for ns, _ := range namespacePVCMap {
+				err := createNameSpace(ns)
+				log.FailOnError(err, "Failed to create namespace [%s]", ns)
+			}
+			for x := 0; x < numberOfPvc; x++ {
+				for storageclass, namespace := range scNamespaceMap {
+					wg.Add(1)
+					go func(x int, storageclass, namespace string) {
+						defer wg.Done()
+						defer GinkgoRecover()
+						createAndAppendPVC(namespace, storageclass, namespace, x)
+					}(x, storageclass, namespace)
+				}
+			}
+			wg.Wait()
+		})
+		//checkPvcBound will check if pvc is bounded or not
+		checkPvcBound := func(listofPvc []string, namespace string) {
```

Comment:
```suggestion
		checkAllPvcBound := func(listofPvc []string, namespace string) {
```

Reply:
Also could we define some of these functions outside of the test function? Some of them feel more like utilities that aren't specific to this test, and it would make the test function itself shorter/more readable.